### PR TITLE
feat task-sender-infinite-proofs-plonk

### PR DESCRIPTION
**Pull request by jotabulacios**
This PR introduces the capability to send infinite PLONK BN254 proofs using both the GO task sender and the batch task sender

To send a different valid plonk bn254 proof every 3 seconds run:
```bash
make send_infinite_plonk_bn254_proof
```

To sends a different valid plonk bn254 proof every n seconds, using batcher task sender  run:
```bash
make batcher_send_infinite_plonk_bn254_tasks
```

In case you want to do it with Groth16, run:

```bash
make batcher_send_infinite_groth16_tasks
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -141,14 +141,24 @@ batcher_send_groth16_task: batcher/client/target/release/batcher-client
 		--public_input ../../test_files/groth16/ineq_1_groth16.pub \
 		--vk ../../test_files/groth16/ineq_1_groth16.vk \
 
-batcher_send_infinite_groth16: ./batcher/client/target/release/batcher-client ## Send a different Groth16 BN254 proof using the task sender every 3 seconds
-	@mkdir -p task_sender/test_examples/gnark_groth16_bn254_infinite_script/infinite_proofs
+batcher_send_plonk_task: batcher/client/target/release/batcher-client
+	@echo "Sending PlonkBn254 1!=0 task to Batcher..."
+	@cd batcher/client/target/release && ./batcher-client \
```
Comment (uri-99) [inline Makefile]: call using cargo run, as in [this PR](https://github.com/yetanotherco/aligned_layer/pull/206/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52).

This makes rust to compile if necesarry

File: Makefile [hunk 2]
```diff
@@ -141,14 +141,24 @@ batcher_send_groth16_task: batcher/client/target/release/batcher-client
...
+		--proving_system GnarkPlonkBn254 \
+		--proof ../../test_files/groth16/ineq_1_groth16.proof \
+		--public_input ../../test_files/plonk_bn254/ineq_1_plonk_bn254.pub \
+		--vk ../../test_files/groth16/ineq_1_plonk_bn254.vk \
+
+batcher_send_infinite_groth16_tasks: ./batcher/client/target/release/batcher-client ## Send a different Groth16 BN254 proof using the task sender every 3 seconds
+	@mkdir -p task_sender/test_examples/gnark_groth16_bn254_infinite_script/infinite_proofs
```
Comment (uri-99) [inline Makefile]: call using cargo run, as in [this PR](https://github.com/yetanotherco/aligned_layer/pull/206/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52).

This makes rust to compile if necesarry

**Review by uri-99 — CHANGES_REQUESTED** [review]

File: Makefile [hunk 3]
```diff
@@ -153,20 +154,30 @@ batcher_send_plonk_bls12_381_task: batcher/client/target/release/batcher-client
...
-		--vk test_files/groth16/ineq_1_groth16.vk
+		--proof ../../test_files/groth16/ineq_1_groth16.proof \
+		--public_input ../../test_files/groth16/ineq_1_groth16.pub \
+		--vk ../../test_files/groth16/ineq_1_groth16.vk \
+
+batcher_send_plonk_task: batcher/client/target/release/batcher-client
+	@echo "Sending PlonkBn254 1!=0 task to Batcher..."
+	@cd batcher/client/ && cargo run --release --  \
+		--proving_system GnarkPlonkBn254 \
+		--proof ../../test_files/groth16/ineq_1_groth16.proof \
```
Comment (uri-99) [inline Makefile]: this .proof should be plonk

**Review by uri-99 — CHANGES_REQUESTED** [review]

File: Makefile [hunk 4]
```diff
@@ -153,20 +154,30 @@ batcher_send_plonk_bls12_381_task: batcher/client/target/release/batcher-client
...
+	@cd batcher/client/ && cargo run --release --  \
 		--proving_system Groth16Bn254 \
-		--proof test_files/groth16/ineq_1_groth16.proof \
-		--public_input test_files/groth16/ineq_1_groth16.pub \
-		--vk test_files/groth16/ineq_1_groth16.vk
+		--proof ../../test_files/groth16/ineq_1_groth16.proof \
+		--public_input ../../test_files/groth16/ineq_1_groth16.pub \
+		--vk ../../test_files/groth16/ineq_1_groth16.vk \
+
+batcher_send_plonk_task: batcher/client/target/release/batcher-client
```
Comment (uri-99) [inline Makefile]: this target should require the files to be generated, or should be stored in the repo

File: batcher/client/send_infinite_tasks.sh [hunk 5]
```diff
@@ -12,16 +14,30 @@ else
...
-    cd ../..
+        ./batcher/client/target/release/batcher-client --proving_system Groth16Bn254 --proof task_sender/test_examples/gnark_groth16_bn254_infinite_script/infinite_proofs/ineq_${counter}_groth16.proof --public_input task_sender/test_examples/gnark_groth16_bn254_infinite_script/infinite_proofs/ineq_${counter}_groth16.pub --vk task_sender/test_examples/gnark_groth16_bn254_infinite_script/infinite_proofs/ineq_${counter}_groth16.vk
+    else
+        go run task_sender/test_examples/gnark_plonk_bn254_infinite_script/cmd/main.go $counter
 
+        ./batcher/client/target/release/batcher-client --proving_system GnarkPlonkBn254 --proof task_sender/test_examples/gnark_plonk_bn254_infinite_script/infinite_proofs/ineq_${counter}_plonk.proof --public_input task_sender/test_examples/gnark_plonk_bn254_infinite_script/infinite_proofs/ineq_${counter}_plonk.pub --vk task_sender/test_examples/gnark_plonk_bn254_infinite_script/infinite_proofs/ineq_${counter}_plonk.vk
```
Comment (uri-99) [inline batcher/client/send_infinite_tasks.sh:38]: these should be executed using cargo run

File: task_sender/cmd/main.go [hunk 6]
```diff
@@ -303,8 +333,21 @@ func taskSenderInfiniteMain(c *cli.Context) error {
 	x := 0
 	for {
 		x += 1
-		generateproof.GenerateIneqProof(x)
-		err := taskSenderMain(c, x)
+		provingSystem, err := ParseProvingSystem(c.String(provingSystemFlag.Name))
+		if err != nil {
+			return err
+		}
```
Comment (uri-99) [inline task_sender/cmd/main.go]: this can go outside the for

**Review by uri-99 — CHANGES_REQUESTED** [review]

File: Makefile (hunk 4 above)
Comment (jotabulacios) [reply Makefile]: Those should be stored in the same manner as the ones from Groth16.

**Pull request closed without merging**